src/modules/csm/csm.py

```diff
@@ -447,3 +508,69 @@ def _get_staking_module(self) -> StakingModule:
                 return mod
 
         raise NoModuleFound
+
+
+@dataclass
+class LastReport:
+    w3: Web3
+    blockstamp: BlockStamp
+
+    rewards_tree_root: HexBytes
+    strikes_tree_root: HexBytes
+    rewards_tree_cid: CID | None
+    strikes_tree_cid: CID | None
+
+    @classmethod
+    def load(cls, w3: Web3, blockstamp: BlockStamp) -> Self:
+        rewards_tree_root = w3.csm.get_rewards_tree_root(blockstamp)
+        rewards_tree_cid = w3.csm.get_rewards_tree_cid(blockstamp)
+
+        if (rewards_tree_cid is None) != (rewards_tree_root == ZERO_HASH):
+            raise InconsistentData(f"Got inconsistent previous tree data: {rewards_tree_root=} {rewards_tree_cid=}")
+
+        strikes_tree_root = w3.csm.get_strikes_tree_root(blockstamp)
+        strikes_tree_cid = w3.csm.get_strikes_tree_cid(blockstamp)
+
+        if (strikes_tree_cid is None) != (strikes_tree_root == ZERO_HASH):
+            raise InconsistentData(f"Got inconsistent previous tree data: {strikes_tree_root=} {strikes_tree_cid=}")
+
+        return cls(
+            w3,
+            blockstamp,
+            rewards_tree_root,
+            strikes_tree_root,
+            rewards_tree_cid,
+            strikes_tree_cid,
+        )
+
+    @cached_property
+    def rewards(self) -> Iterable[RewardsTreeLeaf]:
+        if self.rewards_tree_cid is None or self.rewards_tree_root == ZERO_HASH:
```

Comment:
```suggestion
        if self.rewards_tree_cid is None and self.rewards_tree_root == ZERO_HASH:
```

Reply:
Doesn't make sense for me. Added a check that these two values are empty simultaneously instead.